pkg/data/management/ensure_admin_user.go

```diff
@@ -119,6 +120,18 @@ func createNewAdmin(client v3.Interface, length int) error {
 
 	fmt.Fprintf(os.Stdout, "New default admin user (%v):\n", admin.Name)
 	fmt.Fprintf(os.Stdout, "New password for default admin user (%v):\n%s\n", admin.Name, pass)
+
+	//sleep to avoid "object has been changed" condition
+	time.Sleep(3 * time.Second)
```

Comment:
This is not a safe way to handle these conflict errors. Follow this pattern instead: https://github.com/rancher/rancher/blob/6f9c19dc3e1daa4880c11dee036899f4ad2b5340/pkg/controllers/management/auth/crtb_handler.go#L230-L242

That said, there is nothing happening between the user create on line 104 and the update on line 134, so the admin object returned from Create should be up to date and safe to act on. But if for some reason that's not the case, use RetryOnConflict.

Reply:
I was seeing some errors without the time.sleep - I think there's a controller somewhere that makes edits on the user right after it's created which is causing those issues. I'll migrate to the retryOnConflict method, thanks for the suggestion.

Reply:
Fixed - thanks again for the suggestion.